### PR TITLE
refactor: drop gocsv to stdlib encoding/csv 

### LINKED ISSUE #674: fix: remove gocsv dependency
Remove the github.com/gocarina/gocsv dependency by implementing a local CSV marshaler using stdlib encoding/csv and reflect for struct tags.

**Files:** 
- internal/output/marshal.go:77
- cmd/tuple/read.go:50–58

**Changes:**
- Create a local marshalCSV(data any) helper function using encoding/csv and reflect
- Parse csv struct tags to generate headers and rows
- Replace gocsv.MarshalBytes() call with the new helper

### PR DESCRIPTION
**Pull request by SoulPancake**
## Description

 - Replace gocsv.MarshalBytes with a local reflect-based marshalCSV
    helper using encoding/csv, fully removing the gocsv dependency
    (closes #674)
  - Add marshalCSV test coverage (escaping, omitempty, nil pointers,
    non-string fields, error cases)

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test suite for CSV output formatting, covering edge cases like escaped characters, multi-row output, and error scenarios.

* **Refactor**
  * Improved internal error handling mechanisms and reduced external dependency requirements for better stability and maintainability.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -49,6 +47,7 @@ require (
 	github.com/grpc-ecosystem/go-grpc-middleware/v2 v2.3.3 // indirect
 	github.com/grpc-ecosystem/grpc-gateway/v2 v2.29.0 // indirect
 	github.com/hashicorp/errwrap v1.1.0 // indirect
+	github.com/hashicorp/go-multierror v1.1.1 // indirect
```
[the remark this answers is not in the document]
Comment (SoulPancake) [reply go.mod]: Yes it is still indirect as it is using in the transformer in language/pkg/go

[the remark this answers is not in the document]
Comment (SoulPancake) [reply go.mod]: This is because of pkg/go/transformer
We will maybe drop that and update it later to remove the indirect dep

**Comment by SoulPancake** [comment]
We can maybe get [this](https://github.com/openfga/language/issues/600) done before we update and  merge this
so we can also remove the indirect deps

**Review by rhamzeh — COMMENTED** [review]
I would rather not maintain our own CSV parsing implementation - a battle tested library for this case wins.

We can always move off it if needed, but for this niche case is an exception where I'm willing to tolerate a third-party lib.

At the very least if we can move that off to its own PRs so we can easily revert it.

For the multi error change, let's bump up language after it is merged in the same PR here before merging this.

**Comment by senojj** [comment]
Using reflection to identify struct tags is slow and generally unnecessary for CSV; CSV formatted data records are all of the same shape for a given set, with an optional field name header. Unlike JSON which encodes fields names and other properties of the fields. For CSV encoding, I would advise using an interface along with the standard library instead. Something like:
```go
type Marshaler interface {
    MarshalCSV() ([]string, error)
}
```
And define your struct like:
```go
type Person struct {
    FirstName string
    LastName  string
    Age       int
}

func (p *Person) MarshalCSV() ([]string, error) {
    var output [3]string
    output[0] = p.FirstName
    output[1] = p.LastName
    output[2] = strconv.Itoa(p.Age)
    return output[:], nil
}
```
Then you can have a helper:
```go
func MarshalCSV(records []Marshaler, w io.Writer, header ...string) error {
    ow := csv.NewWriter(w)
    if len(header) > 0 {
        err := ow.Write(header)
        if err != nil {
            return err
        }
    }

    for _, r := range records {
        output, err := r.MarshalCSV()
        if err != nil {
            return err
        }
        err = ow.Write(output)
        if err != nil {
            return err
        }
    }
    return nil
}
```

File: internal/authorizationmodel/model.go [hunk 2]
```diff
@@ -207,16 +207,16 @@ func (model *AuthzModel) ReadModelFromModFGA(modFile string) error {
 	}
 
 	moduleFiles := []language.ModuleFile{}
-	fileReadErrors := multierror.Error{}
+	fileReadErrors := []error{}
```
Comment (senojj) [inline internal/authorizationmodel/model.go]: This incurs an allocation even when no read errors are encountered. Start with a `nil` slice, and append will allocate when needed.

**Comment by SoulPancake** [comment]
Thanks alot @rhamzeh and @senojj for the feedback.
I’ll keep this PR focused on the CSV parsing changes and address the related review comments already here. I’ll move the go-multierror and language pkg bump into a separate PR.

**Comment by SoulPancake** [comment]
I have addressed the comments but I'm honestly okay with keeping the third party library as well! 
However this PR might just help undo it if we ever choose to do it

File: internal/output/marshal.go [hunk 3]
```diff
@@ -84,6 +86,53 @@ func (prt *csvPrinter) DisplayNoColor(data any) error {
...
+}
+
+// CSVRecordSet is implemented by a collection that can render itself as CSV:
+// a header row followed by one record per element.
+type CSVRecordSet interface {
+	CSVHeaders() []string
+	CSVRecords() []CSVMarshaler
+}
+
+func marshalCSV(data any) ([]byte, error) {
```
Comment (senojj) [inline internal/output/marshal.go]: why do you need to accept `any` as the argument type? you seem to require a `CSVRecordSet` in the function body.

Comment (senojj) [reply internal/output/marshal.go]: shift the type assertion to where it is necessary, in `DisplayNoColor`

File: cmd/tuple/read.go [hunk 4]
```diff
@@ -48,18 +48,57 @@ type readResponse struct {
...
+}
+
+func (l readResponseCSVDTOList) CSVRecords() []output.CSVMarshaler {
+	records := make([]output.CSVMarshaler, len(l))
+	for i, dto := range l {
+		records[i] = dto
+	}
+
+	return records
+}
```
Comment (senojj) [inline cmd/tuple/read.go]: This function adds overhead that is entirely unnecessary.

**Comment by senojj** [comment]
My original suggestion put the marshaling interface at the individual object instance level. That allowed for better composition than applying the serialization directly to a container type. Additionally, the original suggestion remained closer to the implementations found within the Go standard library, and thus would be more familiar to Go developers.

I'm not asking for the originally suggested code to be used verbatim, but the spirit of the approach should be maintained.

File: cmd/tuple/read.go [hunk 5]
```diff
@@ -196,19 +221,22 @@ var readCmd = &cobra.Command{
...
 		if outputFormat == "csv" {
-			data, _ := response.toCsvDTO()
-
-			err := dataPrinter.Display(data)
+			records, err := response.toCsvDTO()
 			if err != nil {
+				return fmt.Errorf("failed to convert response to csv: %w", err)
+			}
+
+			if err := output.MarshalCSV(records, os.Stdout, readResponseCSVHeaders...); err != nil {
```
Comment (senojj) [inline cmd/tuple/read.go]: `MarshalCSV` calls `Write` on the provided `io.Writer` multiple times -- as it should. Each call to `Write` on `os.Stdout` is a direct system call, which will be very inefficient. You should initialize a buffered writer, pass the buffer into `MarshalCSV` and write the buffer contents to stdout as a single system call.

Comment (SoulPancake) [reply cmd/tuple/read.go]: Done, Thanks

File: cmd/tuple/read.go [hunk 6]
```diff
@@ -196,19 +222,27 @@ var readCmd = &cobra.Command{
...
 		if outputFormat == "csv" {
-			data, _ := response.toCsvDTO()
-
-			err := dataPrinter.Display(data)
+			records, err := response.toCsvDTO()
 			if err != nil {
+				return fmt.Errorf("failed to convert response to csv: %w", err)
+			}
+
+			var buf bytes.Buffer
```
Comment (senojj) [inline cmd/tuple/read.go]: My apologies for not being more clear with my meaning. I had said, "write to `os.Stdout` using a single system call" and that is what you are doing here. However, for large outputs, accumulating the entire stream in memory could be unwise. Instead of using a `bytes.Buffer` use a `bufio.NewWriter(os.Stdout)` so that the writes are buffered and chunked to `os.Stdout`. Additionally, `csv.NewWriter` can take advantage of a `bufio.Writer` instance, and won't create a new one on top of it.

Comment (SoulPancake) [reply cmd/tuple/read.go]: Oops got it, no worries
used bufio.writer and removed the double wrapping

**Review by senojj — APPROVED** [review]
